.github/workflows/go.yml

```diff
@@ -5,16 +5,18 @@ on:
     branches: [ main ]
   pull_request:
     branches: [ main ]
+  workflow_dispatch:
```

Comment:
what does this do?

Reply:
it permits the Maintainers/Admins of the repo manually trigger the workflow from the UI.
Majority of the time this isn't actually needed. But we've found that it is very useful in exceptional circumstances to be able to trigger a workflow manually.  So it's a 'safety feature' for the future.